okio/src/main/java/okio/Buffer.java

```diff
@@ -589,14 +585,15 @@ public void clear() {
       if (head.pos == head.limit) {
         Segment toRecycle = head;
         head = toRecycle.pop();
-        SegmentPool.INSTANCE.recycle(toRecycle);
+        SegmentPool.recycle(toRecycle);
       }
     }
   }
 
   @Override public Buffer write(ByteString byteString) {
     if (byteString == null) throw new IllegalArgumentException("byteString == null");
-    return write(byteString.data, 0, byteString.data.length);
+    byteString.write(this);
```

Comment:
the evolution of buffer and bytestring together are starting to make this method look deprecate worthy.. except for its chaining virtue.